hikari/impl/rest.py

```diff
@@ -1253,15 +1259,15 @@ async def _create_message(
 
         if undefined.all_undefined(embed, embeds) and isinstance(content, embeds_.Embed):
```

Comment:
What about if embed or embeds is None

Reply:
Also I'm not inferring this is bad behaviour, i'm just asking cause it seems like this is something that previously slipped through the gap without any properconsideration

Reply:
This is also the same as the other comment bellow. This is the syntactic sugar. If they want to pass embed as None, they can do `embed=None` or `embeds=None`. I found while testing that some parts of the code (and typing) said that `component` and `components` could be `None`, but then the check didnt allow it, which I fixed:

https://github.com/hikari-py/hikari/pull/971/files#diff-9904e7f903a912bfe485f6f673fcd1e42da8b90786d58fb612077b2c42b68bfaR1248

Reply:
Yeah but with the current behaviour passing an embed to content then None to embed or embeds leads to the Nones being ignored which is why I asked about the semantics

Reply:
Honestly, I would go with "intended behavior" then. The documentation clearly specifies that the syntactic sugar only triggers if the other arguments are not passed:
```
        content : hikari.undefined.UndefinedOr[typing.Any]
            If provided, the message contents. If
            `hikari.undefined.UNDEFINED`, then nothing will be sent
            in the content. Any other value here will be cast to a
            `builtins.str`.

            If this is a `hikari.embeds.Embed` and no `embed` nor
            no `embeds` kwarg is provided, then this will instead
            update the embed. This allows for simpler syntax when
            sending an embed alone.

            Likewise, if this is a `hikari.files.Resource`, then the
            content is instead treated as an attachment if no `attachment` and
            no `attachments` kwargs are provided.
 ```